pennylane/ops/functions/equal.py

```diff
@@ -208,6 +210,21 @@ def _equal_operators(
     return getattr(op1, "inverse", False) == getattr(op2, "inverse", False)
 
 
+@_equal.register
+# pylint: disable=unused-argument, protected-access
+def _equal_prod_and_sum(op1: CompositeOp, op2: CompositeOp, **kwargs):
+    """Determine whether two Prod or Sum objects are equal"""
+
+    # organizes by wire indicies while respecting commutation relations
+    sorted_ops1 = op1._sort(op1.operands)
+    sorted_ops2 = op2._sort(op2.operands)
+
+    if len(sorted_ops1) != len(sorted_ops2):
```

Comment:
If sorting doesn't change the number of ops, it might be good to run this check first before sorting. That way we don't have to go through the effort of sorting if we can immediately tell that the operations aren't equal.